src/utils.ts

```diff
@@ -38,7 +39,9 @@ export function pluckData<R>(obj: MaybeData<R>): R {
  */
 export function catchNotFoundOrThrow(err: ApifyApiError): void {
     const isNotFoundStatus = err.statusCode === NOT_FOUND_STATUS_CODE;
-    const isNotFoundMessage = err.type === NOT_FOUND_TYPE || err.message.includes(NOT_FOUND_ON_S3);
+    const isNotFoundMessage = err.type === RECORD_NOT_FOUND_TYPE
+        || err.type === RECORD_OR_TOKEN_NOT_FOUND_TYPE
+        || err.message.includes(NOT_FOUND_ON_S3);
```

Comment:
I wonder if we still need this. I think we removed the S3 redirects some time ago, no?

Reply:
You're right, we removed those ages ago. I removed the special S3 error handling too.